cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -98,10 +115,20 @@
   private boolean suspended;
   private Lock lock;
   private Condition condition;
+  private final RunId runId;
+  private final CConfiguration cConf;
+  private final MetricsCollectionService metricsCollectionService;
+  private final DatasetFramework datasetFramework;
+  private final DiscoveryServiceClient discoveryServiceClient;
+  private final TransactionSystemClient txClient;
 
   WorkflowDriver(Program program, ProgramOptions options, InetAddress hostname,
-                 WorkflowSpecification workflowSpec, ProgramRunnerFactory programRunnerFactory) {
+                 WorkflowSpecification workflowSpec, ProgramRunnerFactory programRunnerFactory, RunId runId,
```

Comment:
We can get RunId from the options itself 
`RunId runId = RunIds.fromString(options.getArguments().getOption(ProgramOptionConstants.RUN_ID));`
No need to pass in the constructor.